pkg/pipeline/conntrack/conntrack.go

```diff
@@ -86,24 +154,45 @@ func (ct *conntrackImpl) Track(flowLogs []config.GenericMap) []config.GenericMap
 			outputRecords = append(outputRecords, fl)
 		}
 	}
+
+	endConnectionRecords := ct.popEndConnections()
+	if ct.shouldOutputEndConnection {
+		outputRecords = append(outputRecords, endConnectionRecords...)
+	}
+
 	return outputRecords
 }
 
-func (ct *conntrackImpl) addConnection(hashStr string, conn connection) {
-	ct.hash2conn[hashStr] = conn
+func (ct *conntrackImpl) popEndConnections() []config.GenericMap {
+	var outputRecords []config.GenericMap
+	ct.connStore.iterateOldToNew(func(conn connection) (shouldDelete, shouldStop bool) {
```

Comment:
This need to be async based on the timeout of the connection and not "just" based on the assumption that we get once in a while a new flow-log that will trigger.

@mariomac is this doable? do we need to use your channel PR for this to happen ???